### PR TITLE
docs: clarify usage of upsert with surrogate primary keys

### PR DESCRIPTION
**Pull request by laurenceisla**
Clarify the Upsert docs according to this conversation: https://github.com/PostgREST/postgrest/issues/1118#issuecomment-2342545959

Also changed the example to one that uses natural primary keys instead.

### CODE REVIEW

**Comment by elyobo** [comment]
An example of how one would do this with surrogate keys as well would probably be helpful.

**Comment by laurenceisla** [comment]
> Is the information in the issue still correct? I remember we added something about handling DEFAULT values instead of NULL for missing values at some point.

Thanks for the heads up, you're absolutely right! e.g. on `GENERATED BY DEFAULT...` or `SERIAL` PK columns we can add both preferences `Prefer: missing=default, resolution=merge-duplicates` and specify the `columns` to make this work (no need to link the issue anymore). I'll modify accordingly.

Also, I don't see an Upsert spec test that mixes both preferences, I'll add one too.

File: test/spec/Feature/Query/UpsertSpec.hs [hunk 1]
```diff
@@ -103,6 +103,32 @@ spec =
...
             }
 
+        it "INSERTs and UPDATEs rows with SERIAL surrogate primary keys using Prefer: missing=default" $
+          request methodPost "/surr_serial_upsert?columns=id,name" [("Prefer", "return=representation"), ("Prefer", "resolution=merge-duplicates"), ("Prefer", "missing=default")]
+            [json| [
+              { "id": 1, "name": "updated value" },
+              { "name": "new value" }
+            ]|] `shouldRespondWith` [json| [
+              { "id": 1, "name": "updated value" },
+              { "id": 3, "name": "new value" }
```
[the remark this answers is not in the document]
Comment (laurenceisla) [reply test/spec/Feature/Query/UpsertSpec.hs]: True. I added an extra column and selected it instead (if `null` value then it's new, if existing value then it's updated), just to be extra safe.

**Comment by laurenceisla** [comment]
@elyobo I [added an example](https://docs.postgrest.org/en/latest/references/api/tables_views.html#upsert) on what to do with surrogate keys. There is no need for a TRIGGER.

**Pull request merged**